### ISSUE TITLE
Wire survey up to rtk-query

### ISSUE THREAD

**Issue by mattdelsordo** [codeforboston/urban-league-heat-pump-accelerator#334]
See #325 for the related back end change.

Survey payload to the back end should look something like the following:
```
{
    "survey_visit": {
        "surveyor_id": 1,
        "home_id": 1,
        "survey_response_attributes": {
            "survey_id": 1,
            "completed": false,
            "survey_answers_attributes": {
                "1": {
                    "survey_question_id": 1,
                    "answer": "Yes"
                },
                "2": {
                    "survey_question_id": 2,
                    "answer": "No"
                }
            }
        }
    }
}
```
Update the jsonserver data accordingly because it is likely out of date. `surveyor_id` should be null in the public view case.

For ReCAPTCHA support, back end is working on that here: #327
I think they said they wanted the token and action to be passed in as headers, NOT in the body.

**Comment by mattdelsordo** [comment]
Recaptcha token should be passed in in the `Recaptcha-Token` header as per https://github.com/codeforboston/urban-league-heat-pump-accelerator/pull/327/files#diff-b728913246a2ea557bfcbc72f058e64642e1e4de44b053198fda2a025cc55e73R38

**Comment by mikelynch** [comment]
That's right. I think we thought that when you're looking at a home, you can choose to do a survey – I don't think we saw this as being a different view for the surveyors or public. When you do the survey (or as a surveyor, log that you couldn't do a survey), you're submitting the object like the nested JSON above. We're only interested in doing the reCAPTCHA stuff **for anonymous users** (i.e. a logged-in surveyor doesn't it, as we'll assume those are trusted).

**Comment by mattdelsordo** [comment]
Yep, currently submitting a survey does an API call but it's not POSTing the correct data (and I'm not sure to the correct endpoint either).

**Comment by LWest001** [comment]
@mattdelsordo Issues I'm running in to:
* Can't submit a survey with `surveyor_id: null` or a dummy value like `surveyor_id: public`. This field must contain an id of a real surveyor (server response `{"surveyor":["must exist"]}`).
* Can't obtain a ReCaptcha token (receiving error `Error obtaining ReCAPTCHA token!` as defined in ReCaptcha.js). Nevertheless, Surveys are submitting through the public view in my branch.
* Not sure if all the data is being added to survey_visits and survey_responses properly on submission.

**Comment by mattdelsordo** [comment]
> Can't submit a survey with surveyor_id: null or a dummy value like surveyor_id: public. This field must contain an id of a real surveyor (server response {"surveyor":["must exist"]}).

That's not good, I think that field might need to be changed to optional or whatever on the back end, we should bring this up with them.

> Can't obtain a ReCaptcha token (receiving error Error obtaining ReCAPTCHA token! as defined in ReCaptcha.js). Nevertheless, Surveys are submitting through the public view in my branch.

Oh you probably don't have the API key! I'll PM it to you.

> Not sure if all the data is being added to survey_visits and survey_responses properly on submission.
iirc Randy figured out how to inspect the database, but also if the survey_visit is created properly I think we can trust in the back end team that all the sub-entities are being created as well.

**Comment by LWest001** [comment]
> iirc Randy figured out how to inspect the database, but also if the survey_visit is created properly I think we can trust in the back end team that all the sub-entities are being created as well.

The sub-entities are being created, but some of the attributes don't seem to be included when I query them (eg `completed` for survey_responses or `home_id` for survey_visits.

**Comment by LWest001** [comment]
I got recaptcha wired up properly, with token attached as a header using the action `create_survey`. 

Last little problem here is deriving the `surveyor_id`. There is no surveyor id in the redux store so my only option would be to work up the chain from the provided `home_id` and provide one surveyor from the array of `surveyor_ids` in an assignment. Something like this:
```
const { data: homeData, [...] } = useGetHomeQuery(homeId);
const assignmentId = homeData?.assignment_id;
const { data: assignmentData,  [...] } = useGetAssignmentQuery(assignmentId);
const surveyorIds = assignmentData?.surveyor_ids;

   const surveyVisit = await addSurveyVisit({
        surveyVisit: {
          survey_visit: {
            surveyor_id: surveyorIds[0],
            [...],
          },
        },
      }); 
```
It's just not really accurate because one of two surveyors could be responsible for submitting the survey, or, who knows, maybe some other substitute surveyor performs the action. It's also a little bit convoluted working up the chain this way. I think it should be pulled from login data from the redux store if possible.

I'll submit a PR so we can look at this more in depth.

**Comment by mattdelsordo** [comment]
Re: the surveyor ID that's really an auth problem, it'll sort of be fixed after #318 is merged but what we really need is #322 so that the surveyor entity associated with a user entity is returned directly

**Comment by LWest001** [comment]
> > Can't submit a survey with surveyor_id: null or a dummy value like surveyor_id: public. This field must contain an id of a real surveyor (server response {"surveyor":["must exist"]}).
> 
> That's not good, I think that field might need to be changed to optional or whatever on the back end, we should bring this up with them.

This should be repaired when we merge branch [190-add-homes-public-survey](https://github.com/codeforboston/urban-league-heat-pump-accelerator/tree/190-add-homes-public-survey).
